### PR TITLE
gpmdp: new module from scratch

### PR DESCRIPTION
**Pull request by lasers**
We revives a closed pull request. https://github.com/ultrabug/py3status/pull/878

* support 5+ external placeholders, same as before.
* remove dependency `gpmdp-remote`
* remove code where it'll run few things per placeholder on every loop
    1) it runs loop calling `gpmdp-shell` for every placeholder.
    2) shell `gpmdp-remote` itself contains `awk` and two `printf`
    3) instead of all that, we can just parse a json file ourselves in python.

Minor changes by taking advantage of new `allow` conditions...
* no more `is_playing`, `is_paused`, and `is_stopped`.
    1) use `\?if=playing` (for playing/paused) and `\?if=!title` (for stopped).
* i remove boolean placeholders for `repeat` and `shuffle`.
   1) users could do `\?if=!repeat=NO_REPEAT` (not sure if this is correct).
   2) users could do `\?if=!shuffle=NO_SHUFFLE` (not sure if this is correct).
* no more hardcoded colors. them colors in the `format` now.

Things to know...
* Get rid of deprecation? Original PR say yes?
* I keep `is_started` as there is no way to tell if `gpmdp` is running (so i can hide the module when it's not running).
* I don't like this...
    ```
    +    {time_currenttime} current time in [HH:]MM:SS
    +    {time_current} current time in milliseconds
    +    {time_totaltime} total time in [HH:]MM:SS
    +    {time_total} total time in milliseconds
    ```
   1) do we override `time_current` + `time_total` or use new formatter? 
* `gpmdp` client itself partially work with `mpris`. when clicking on `mpris`, `mpris` will throw an error. thx.

### CODE REVIEW

File: py3status/modules/gpmdp.py [hunk 1]
```diff
@@ -3,60 +3,148 @@
...
 
 class Py3status:
     """
     """
     # available configuration parameters
     cache_timeout = 5
-    format = u'♫ {info}'
+    format = '[\?if=is_started [\?if=title [\?if=playing&color=playing ' +\
+        '{title} - {artist}|\?color=paused (Paused) {title} - {artist}]|' +\
+        '\?color=stopped Google Play Music]]'
```
Comment (tobes) [inline py3status/modules/gpmdp.py]: I detest the `\` line extension it always reminds me of VB which causes psychotic episodes.  There is a little trick in python where strings will append to each other.  It is also the cause of some of the hardest bugs I've tried to find in python but you can do this

```python
    format = ('[\?if=is_started [\?if=title [\?if=playing&color=playing '
        '{title} - {artist}|\?color=paused (Paused) {title} - {artist}]|'
        '\?color=stopped Google Play Music]]')
```

Comment (lasers) [reply py3status/modules/gpmdp.py]: :white_check_mark:

**Comment by tobes** [comment]
@lasers once again sorry I've not been around much of late.  I'll try to be a bit more active once 3.7 is released so we can try to close some of these long term PRs

Realistically we'll have to split them up into the nice non-controversial bits so we can get some movement happening and then the more difficult bits will become much clearer to negotiate on.  

And thanks for your continued contributions

**Comment by lasers** [comment]
Sure. It was hard to keep going nonstop. Let me know when you'd like to go through them. We can have a disucssion too and I can give brief statement on each PR. Some thing is nice, some thing is meh, some thing is just ready. We should try and get `py3status` in many many linux distributions too (officially).

**Comment by lasers** [comment]
Small explanation: I was cleaning out remote branches. Went from 85~ branches down to 32. That's 53~ branches deleted. I also wiped out all 30~ local git stashes. I plan to clean out local branches next. Anyhow, this was a happy accident. One less thing to merge.

**Pull request closed without merging**